packages/bitcore-wallet-service/src/lib/server.ts

```diff
@@ -5029,6 +5029,32 @@ export class WalletService {
     });
   }
 
+  checkServiceAvailability(req): Promise<boolean> {
+    return new Promise((resolve, reject) => {
```

Comment:
I'd like to move away from this sort of verbosity. Could you instead just make this function an `async` function and throw/return below?

Reply:
Fixed. I avoided using promises for this particular function